GAE/src/org/akvo/flow/rest/ImageUploadRestService.java

```diff
@@ -0,0 +1,183 @@
+/*
+ *  Copyright (C) 2021 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+package org.akvo.flow.rest;
+
+import com.gallatinsystems.common.util.S3Util;
+import com.gallatinsystems.survey.dao.QuestionDao;
+import com.gallatinsystems.survey.dao.SurveyDAO;
+import com.gallatinsystems.survey.domain.Question;
+import com.gallatinsystems.survey.domain.Survey;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+import org.waterforpeople.mapping.dao.QuestionAnswerStoreDao;
+import org.waterforpeople.mapping.dao.SurveyInstanceDAO;
+import org.waterforpeople.mapping.domain.QuestionAnswerStore;
+import org.waterforpeople.mapping.domain.SurveyInstance;
+
+@Controller
+@RequestMapping("/image_upload")
+public class ImageUploadRestService {
+
+    private static final Logger log = Logger
+            .getLogger(ImageUploadRestService.class.getName());
+
+    @RequestMapping(method = RequestMethod.POST, value = "/question/{questionId}/instance/{instanceId}")
+    @ResponseBody
+    public Response uploadImage(@PathVariable("questionId") String questionId,
```

Comment:
I believe its possible to already convert to a `Long` here.. Same for the `formInstanceId`

Reply:
```suggestion
    public Response uploadImage(@PathVariable("questionId") Long questionId,
```

Reply:
done